src/Appium.Net/Appium/AppiumOptions.cs

```diff
@@ -8,38 +11,160 @@ namespace OpenQA.Selenium.Appium
+            get { return "appium"; }
+        }
+
+        /// <summary>
+        /// Gets or sets the AutoamtionName of the Appium browser's (e.g. Appium, Selendroid and so on) setting.
+        /// </summary>
+        public string AutomationName  {get; set;}
+
+
+        /// <summary>
+        /// Gets or sets the DeviceName of the Appium browser's (e.g. Pixel 3XL, Galaxy S20 and so on) setting.
+        /// </summary>
+        public string DeviceName  {get; set;}
+
+        /// <summary>
+        /// Gets or sets the Capability name used for the apllication setting.
+        /// </summary>
+        public string App {get; set;}
+
+        /// <summary>
+        /// Gets or sets the PlatformVersion of the Appium browser's (e.g. 10, 11 and so on) setting.
+        /// </summary>
+        public string PlatformVersion { get; set; }
+
+        /// <summary>
+        /// Provides a means to add additional capabilities not yet added as type safe options
+        /// for the Appium driver.
+        /// </summary>
+        /// <param name="optionName">The name of the capability to add.</param>
+        /// <param name="optionValue">The value of the capability to add.</param>
+        /// <exception cref="ArgumentException">
+        /// thrown when attempting to add a capability for which there is already a type safe option, or
+        /// when <paramref name="optionName"/> is <see langword="null"/> or the empty string.
+        /// </exception>
+        /// <remarks>Calling <see cref="AddAdditionalAppiumOption(string, object)"/>
+        /// where <paramref name="optionName"/> has already been added will overwrite the
+        /// existing value with the new value in <paramref name="optionValue"/>.
+        /// Calling this method adds capabilities to the Appium-specific options object passed to
+        /// webdriver executable.</remarks>
+        public void AddAdditionalAppiumOption(string optionName, object optionValue)
+        {
+            this.ValidateCapabilityName(optionName);
+
+            string name;
+
+            if (optionName.Contains(":"))
+            {
+                name = optionName;
+            }
+            else
             {
-                throw new ArgumentException("Capability name may not be null an empty string.", "capabilityName");
+                name = $"{this.VendorPrefix}:{optionName}";
             }
 
-            this.capabilities[capabilityName] = capabilityValue;
+            this.ValidateCapabilityName(name);
+            this.additionalAppiumOptions[name] = optionValue;
         }
 
         /// <summary>
+        /// Add new capabilities
+        /// </summary>
+        /// <param name="capabilityName">Capability name</param>
+        /// <param name="capabilityValue">Capabilities value, which cannot be null or empty</param>
+        [Obsolete("Use the temporary AddAdditionalOption method or the AddAdditionalChromeOption method for adding additional options")]
```

Comment:
Because the natural, “correct” name of the method _should_ be `AddAdditionalCapability`, to be consistent with other language bindings, the spec, documentation, and user expectations set over years of use. However, it wouldn’t be fair to users to remove an overload of the method that is in wide use nor to blindly change behavior of the method immediately with 4.0. I _want_the method to be `AddAdditionalCapability`, and expect to return to that syntax once there has been enough notice to users that the behavior has changed.

Reply:
cannot we just have both methods then and document the one as preferable? I don't really like to put a "time bomb" there, which will explode as soon as this temporary method would be deprecated.

Reply:
@mykola-mokhnach Appium, of course, is free to sculpt its API however it desires. Selenium, however, will be deprecating and renaming the methods at a future time. What will happen is this:
1. At some point, the `Obsolete` tag will be removed from `AddAdditionalCapability`, and added to `AddAdditionalOption`.
2. The `AddAdditionalOption` method will be removed from the Selenium language bindings.
The timeframe for each of these steps will be several minor releases during the 4.x release cycle.
The problem with `AddAdditionalCapability` today is that there are two overloads:
* `AddAdditionalCapability(string name, object value)`, defined in the base `DriverOptions` class, which adds the capability to the browser-specific capabilities (`goog:chromeOptions`, `moz:firefoxOptions`, etc.), and
* `AddAdditionalCapability(string name, object value, bool isGlobalCapability)`, defined in each browser-specific options class, which lets the user tell the bindings to add it to the top-level capabilities.

The boolean value logic is backward from what it ought to be to be put on the base class, which is the reason for this whole dance in the first place, to avoid breaking existing code, rather than silently changing the behavior of `AddAdditionalCapability` under the covers.